### PR TITLE
Add Active Directory and kubelogin integration sample.

### PR DESCRIPTION
**Pull request by okamototk**
This PR includes Active Directory and kubelogin integration sample.

### CODE REVIEW

**Comment by ericchiang** [comment]
Would you mind adding instructions here for how to use this? https://github.com/dexidp/dex/blob/master/Documentation/connectors/ldap.md#getting-started

File: Documentation/connectors/kubelogin-activedirectory.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+```
+$ kubectl config set-context oidc-ctx --cluster=cluster.local --user=test
+$ kubectl config set-credentials test 
+  --auth-provider=oidc \
+  --auth-provider-arg=idp-issuer-url=https://dex.example.com:32000/dex \
+  --auth-provider-arg=client-id=kubernetes \
+  --auth-provider-arg=client-secret=ZXhhbXBsZS1hcHAtc2VjcmV0 \
+  --auth-provider-arg=idp-certificate-authority-data=$(base64 -w 0 openid-ca.pem) \
+  "--auth-provider-arg=extra-scopes=offline_access openid profile email group"
```
Comment (srenatus) [inline Documentation/connectors/kubelogin-activedirectory.md]: would this be the same as 
```suggestion
  --auth-provider-arg="extra-scopes=offline_access openid profile email group"
```
? I suppose it doesn't matter, the latter just looks a bit more natural to me. Feel free to ignore 😉

File: Documentation/connectors/ldap.md [hunk 2]
```diff
@@ -284,3 +284,42 @@ connectors:
 If the search finds an entry, it will attempt to use the provided password to bind as that user entry.
 
 [openldap]: https://www.openldap.org/
+
+## Example: Searching a Active Directory server with groups
+
+The following configuration will allow the LDAP connector to search a Active Directory using an LDAP filter.
+
+```yaml
+
```
Comment (srenatus) [inline Documentation/connectors/ldap.md]: [nit] extra newline at the top and the bottom

File: Documentation/connectors/kubelogin-activedirectory.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+Please confirm idp-issuer-url, cleint-id, client-secret and idp-certificate-authority-data value is same as config-ad-kubelogin.yaml's value.
+
+Then run kubelogin:
+
+```
+$ kubelogin
+```
+
+Access http://localhost:8000 by web browser and login with your AD account(eg. test@example.com) and password.
```
Comment (srenatus) [inline Documentation/connectors/kubelogin-activedirectory.md]:
```suggestion
Access http://localhost:8000 by web browser and login with your AD account (eg. test@example.com) and password.
```

File: Documentation/connectors/kubelogin-activedirectory.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  --auth-provider=oidc \
+  --auth-provider-arg=idp-issuer-url=https://dex.example.com:32000/dex \
+  --auth-provider-arg=client-id=kubernetes \
+  --auth-provider-arg=client-secret=ZXhhbXBsZS1hcHAtc2VjcmV0 \
+  --auth-provider-arg=idp-certificate-authority-data=$(base64 -w 0 openid-ca.pem) \
+  "--auth-provider-arg=extra-scopes=offline_access openid profile email group"
+$ kubectl config use-context oidc-ctx
+```
+
+Please confirm idp-issuer-url, cleint-id, client-secret and idp-certificate-authority-data value is same as config-ad-kubelogin.yaml's value.
```
Comment (srenatus) [inline Documentation/connectors/kubelogin-activedirectory.md]:
```suggestion
Please confirm idp-issuer-url, client-id, client-secret and idp-certificate-authority-data value is same as config-ad-kubelogin.yaml's value.
```

File: Documentation/connectors/kubelogin-activedirectory.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+subjectAltName = @alt_names
+
+[alt_names]
+DNS.1 = dex.example.com
+```
+
+Please replace dex.example.com to your favorit hostname.
+Generate certificate and private key by following command.
+
+```
```
Comment (srenatus) [inline Documentation/connectors/kubelogin-activedirectory.md]: hmm can we use
```
```console
```
here? I think it'll improve the markup:
```console
$ openssl req -new -x509 -sha256 -days 3650 -newkey rsa:4096 -extensions v3_req -out openid-ca.pem -keyout openid-key.pem -config req.cnf -subj "/CN=kube-ca" -nodes
$ ls openid*
openid-ca.pem openid-key.pem
```

File: Documentation/connectors/kubelogin-activedirectory.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+[ v3_req ]
+basicConstraints = CA:FALSE
+keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+subjectAltName = @alt_names
+
+[alt_names]
+DNS.1 = dex.example.com
+```
+
+Please replace dex.example.com to your favorit hostname.
```
Comment (srenatus) [inline Documentation/connectors/kubelogin-activedirectory.md]: [typo]
```suggestion
Please replace dex.example.com to your favorite hostname.
```
Also, I guess "favorite" isn't really what we mean here, is it? 😉

File: Documentation/connectors/ldap.md (hunk 2 above)
Comment (okamototk) [reply Documentation/connectors/ldap.md]: Thanks, fixed.

**Pull request merged**